File: Interpreter/Token.java
package calculator.xwg;

public class Token
{
    enum EType
    {
        NoType,
        Operator,
        Parenthese,
        Comma,
        Number,
        Parameter,
        FunctionName
    }

    Token(EType type, String content){
    	mType = type;
    	mContent = content;
    }
    
    Token(String content){
    	mType = EType.NoType;
    	mContent = content;
    }
    EType getType(){
    	return mType;
    }
    String getContent(){
    	return mContent;
    }
    void setType(EType type){
    	mType = type;
    }
    boolean isNoType(){
    	return (mType == EType.NoType);
    }
    void setContent(String content){
    	mContent = content;
    }
    EType mType;
    String mContent;
};

File: Interpreter/Token.py
from enum import *

class TokenType(Enum):
    NoType = 1
    Operator = 2
    Parenthese = 3
    Comma = 4
    Number = 5
    Parameter = 6
    FunctionName = 7


class Token:
    def __init__(self, p1, p2=None):
        if p2 is None:
            self.mType = TokenType.NoType
            self.mContent = p1
        else:
            self.mType = p1
            self.mContent = p2

    def getType(self):
        return self.mType

    def getContent(self):
        return self.mContent

    def setType(self, type):
        self.mType = type

    def isNoType(self):
        return (self.mType == EType.NoType)

    def setContent(self, content):
        self.mContent = content
